mbed_devices/_internal/htm_file.py

```diff
@@ -0,0 +1,102 @@
+    <title>mbed Website Shortcut</title>
+    </head>
+    <body></body>
+    </html>
+
+---
+
+    <!doctype html>
+    <!-- mbed Platform Website and Authentication Shortcut -->
+    <html>
+    <head>
+    <meta charset="utf-8">
+    <title>mbed Website Shortcut</title>
+    </head>
+    <body>
+    <script>
+    window.location.replace("https://mbed.org/device/?code=460000000988254a00000000000000000000000097969902?version=0253?target_id=00000000000000000000000000000000");
+    </script>
+    </body>
+    </html>
+
+---
+
+    <!-- mbed Microcontroller Website and Authentication Shortcut -->
+    <html>
+    <head>
+    <meta http-equiv="refresh" content="0; url=http://mbed.org/start?auth=101000000000000000000002F7F35E602eeb0bb9b632205c51f6c357aeee7bc9&loader=11972&firmware=16457&configuration=4" />  # noqa
+    <title>mbed Website Shortcut</title>
+    </head>
+    <body></body>
+    </html>
+"""
+import pathlib
+import re
+from typing import Type, TypeVar, Optional, List, Match
+
+
+T = TypeVar("T", bound="HTMFileContentsParser")
+
+
+class HTMFileContentsParser:
+    """Extracts information from MBED.HTM file contents."""
+
+    def __init__(self, file_contents: str) -> None:
+        """Initialises HTMFileParser with file contents."""
+        self._file_contents = file_contents
+
+    @property
+    def target_id(self) -> Optional[str]:
+        """Returns target parsed from the file contents, None if not found."""
+        regex = r"""
+            (?:code|auth)=            # attribute name
+            (?P<target>[a-fA-F0-9]+)  # target identifier
+        """
+        match = _search([regex], self._file_contents, re.VERBOSE)
+        if match:
+            return match["target"]
+        return None
+
+    @property
+    def daplink_version(self) -> Optional[str]:
+        """Returns version parsed from the file contents, None if not found."""
+        as_semver = r"<!--.*Version: (?P<version>\d+\.\d+\.\d+)\.?.*-->"
+        as_digits = r"<!--.*Version: (?P<version>\d+).*-->"
+        match = _search([as_semver, as_digits], self._file_contents)
+        if match:
+            return match["version"]
+        return None
+
+    @property
+    def daplink_build(self) -> Optional[str]:
+        """Returns build num/date parsed from file contents, None if not found."""
+        as_digits = r"<!--.*build (?P<build>\d+).*-->"
+        as_date = r"<!--.*Build: (?P<build>.* \d{2}:\d{2}:\d{2}).*-->"
+        match = _search([as_digits, as_date], self._file_contents)
+        if match:
+            return match["build"]
+        return None
+
+    @classmethod
+    def from_file(cls: Type[T], file_path: str) -> T:
+        """Initialises HTMFileParser with file contents read from given path."""
+        return cls(file_contents=pathlib.Path(file_path).read_text())
+
+
+def _search(regexes: List[str], text: str, flags: int = 0) -> Optional[Match[str]]:
+    for regex in regexes:
+        match = re.search(regex, text, flags)
+        if match:
```

Comment:
Should we just return `match` instead of checking it first? The caller will always have to check the result anyway as this wrapper can return `None`.